packages/react-components/src/InputNumber.tsx

```diff
@@ -240,7 +243,9 @@ export default function InputNumber (props: Props): React.ReactElement<Props> {
 
     if (event.key.length === 1 && !isPreKeyDown) {
       const { selectionStart: i, selectionEnd: j, value } = event.target as HTMLInputElement;
-      const newValue = `${value.substring(0, i || 0)}${event.key}${value.substring(j || 0)}`;
+      let newValue = `${value.substring(0, i || 0)}${event.key}${value.substring(j || 0)}`;
+      // The following will just remove the 0's on the left and send it to check the regex for decimal and allow user to enter number without having to enter backspace
+      newValue = new BN(newValue).toString();
```

Comment:
the second value gonna override the first one, which equals to `const newValue = new BN(newValue).toString();`?